Add control server for taking commands from Android app #11     |

GrubeCube opened the pull request:
A control server that can communicate with the Android
application to take commands for movement and collection is
added. Currently, the server only handles test requests, but is
designed to implement more requests.

In order for the devices to communicate, both must be on the
same router. If disconnection becomes a consistent issue, then
options for communication outside the LAN should be explored.

Server provides basic functionality and issue should be
revisited to implement other command types.

GrubeCube committed:
Add control server for taking commands from Android app #11     |

A control server that can communciates with the Android
application to take commands for movement and collection is
added. Currently, the server only handles test requests, but is
designed to implement more requests.

In order for the devices to communicate, both must be on the
same router. If disconnection becomes a consistent issue, then
options for communciation outside the LAN should be explored.

nb-school commented on server.py:
```diff
@@ -0,0 +1,82 @@
+    CMD_ROTATE_RIGHT_STOP                                       = 105
+    CMD_ROTATE_LEFT_START                                       = 106
+    CMD_ROTATE_LEFT_STOP                                        = 107
+    CMD_COLLECT                                                 = 200
```
I know this is just coding style but do those need to be tabbed so far over? Tbh I did not know this would run without errors in python because of how important spacing is in python syntax.

GrubeCube replied:
I tabbed them over so far to leave room for other (longer worded) commands. Since we only plan on these so far, it's not necessary to have them over. I do prefer them to be lined up though.

GrubeCube replied:
Commit  3122d08b84f914cacb9d71c4dc32b8d1b5c60c8d decreased tab spaces between variables and their initialization. Still kept numbers lined up correctly.

nb-school commented on server.py:
```diff
@@ -0,0 +1,82 @@
+
+def main():
+    # must be ran with 'sudo' to run on port 80
+    httpd = HTTPServer(('', 80), ControlHandler)
```
This is for Apache, correct? Did we have to install Apache on the Pi for this to work?

GrubeCube replied:
I did not have to install any extra libraries. Either it came preinstalled on the Pi or with the Python installation.

nb-school commented on server.py:
```diff
@@ -0,0 +1,82 @@
+
+    def do_GET(self):
+        # deny request
+        self.send_error(418)
```
If you want to keep style consistent these errors should be added with the others above. This does not have to be done though, only if we have time in the project.

GrubeCube replied:
It's a specific HTTP error that I'm sending back, not one I arbitrarily made. Though, I could add a simple constant to keep with styling.

GrubeCube replied:
Commit 3122d08b84f914cacb9d71c4dc32b8d1b5c60c8d removed all magic numbers including HTTP response codes and port. The numbers are now in a global variable with an appropiate name to increase readability.

nb-school reviewed (approved):
For test commands it looks good. I will approve, the code looks good, Just read my comments.

GrubeCube committed:
Small formatting and magic number removal #11

Small formatting changes with spaces to make it easier to read
and magic number removal for more descriptive code.
